src/core/database.cpp

```diff
@@ -1239,6 +1245,18 @@ void Database::AddNextVolumeAsset(int image_asset_id,  wxString name, wxString f
 	AddToBatchInsert("ittiriiitt", image_asset_id, name.ToUTF8().data(), filename.ToUTF8().data(), reconstruction_job_id, pixel_size, x_size, y_size, z_size, half_map_1_filename.ToUTF8().data(), half_map_2_filename.ToUTF8().data());
 }
 
+#ifdef EXPERIMENTAL
+void Database::BeginAtomicCoordinatesAssetInsert()
+{
+	BeginBatchInsert("ATOMIC_COORDINATES_ASSETS", 11, "ATOMIC_COORDINATES_ASSET_ID", "NAME", "FILENAME", "SIMULATION_3D_JOB_ID", "X_SIZE", "Y_SIZE", "Z_SIZE", "PDB_ID", "PDB_AVG_BFACTOR", "PDB_STD_BFACTOR", "EFFECTIVE_WEIGHT");
+}
+
+void Database::AddNextAtomicCoordinatesAsset(int image_asset_id,  wxString name, wxString filename, int simulation_3d_job_id, int x_size, int y_size, int z_size, wxString pdb_id, float pdb_avg_bfactor, float pdb_std_bfactor, float effective_weight )
```

Comment:
So the answer to this and the other non (panelParent) questions is the same: I just followed the recipe from VolumeAssetPanel.

It may well be that simulation_3d_job_id doesn't end up doing anything and may be removed. For now, it is a direct replacement for reconstruction_job_id from VolumeAssetPanel. I didn't want to leave it the same so it would be obvious if I need to change it later.

As for the primary key question, I'm not sure of the original logic, but I've always been a fan of being verbose to maintain control. Maybe there is a good reason @timothygrant80 has this in place? Is there any benefit to having sqlite provide the key?

Reply:
modifying Database::AddNextAtomicCoordinatesAsset to take an AtomicCoordinatesAsset pointer. ( I suppose I could pass by reference as well, but following suite with existing examples, e.g./ AddNextParticlePositionAsset